commcare/src/main/java/org/motechproject/commcare/tasks/builder/model/CaseTypeWithApplicationName.java

```diff
@@ -0,0 +1,57 @@
+package org.motechproject.commcare.tasks.builder.model;
+
+import java.util.List;
+
+/**
+* Stores casetype and it's application name.
+ */
+
+public class CaseTypeWithApplicationName {
+    private String caseType;
+    private String applicationName;
+    private List<String> caseProperties;
+
+    public CaseTypeWithApplicationName(String caseType, String applicationName, List<String> caseProperties) {
+        this.caseType = caseType;
+        this.applicationName = applicationName;
+        this.caseProperties = caseProperties;
+    }
+
+    public String getCaseType() {
+        return caseType;
+    }
+
+    public void setCaseType(String caseType) {
+        this.caseType = caseType;
+    }
+
+    public String getApplicationName() {
+        return applicationName;
+    }
+
+    public void setApplicationName(String applicationName) {
+        this.applicationName = applicationName;
+    }
+
+    public List<String> getCaseProperties() {
+        return caseProperties;
+    }
+
+    public void setCaseProperties(List<String> caseProperties) {
+        this.caseProperties = caseProperties;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Why aren't all fields included in the equals and hashcode methods? On a side note, you can use java7's Objects.equals to reduce complexity and retain the same effect